src/mesh/coordinates.cxx

```diff
@@ -1344,14 +1737,15 @@ Field3D Coordinates::Div_par(const Field3D& f, CELL_LOC outloc,
 // second parallel derivative (b dot Grad)(b dot Grad)
 // Note: For parallel Laplacian use Laplace_par
 
-Field2D Coordinates::Grad2_par2(const Field2D& f, CELL_LOC outloc,
-    const std::string& method) {
+Coordinates::FieldMetric Coordinates::Grad2_par2(const Field2D& f, CELL_LOC outloc,
+                                                       const std::string& method) {
   TRACE("Coordinates::Grad2_par2( Field2D )");
   ASSERT1(location == outloc || (outloc == CELL_DEFAULT && location == f.getLocation()));
 
-  Field2D sg = sqrt(g_22);
-  Field2D result = DDY(1. / sg, outloc, method) * DDY(f, outloc, method) / sg
-                   + D2DY2(f, outloc, method) / g_22;
+  auto invSg = 1.0 / sqrt(g_22);
+  communicate(invSg);
```

Comment:
From looking at the commit history, it looks like this was part of a file-wide change from `Mesh::communicate` to `communicate` to avoid a circular call to `Coordinates` in its constructor. But the previous commit 6259c36 is more illuminating:

> Avoid falling back on to/from field aligned in Grad_par routines

So this should probably be reverted to 

```suggestion
  localmesh->communicate(invSg);
```